utils/codegen/asn1ToC/asn1ToC.py

```diff
@@ -168,62 +168,72 @@ def main():
-            for f in glob.glob(os.path.join(container_output_dir, "*.h")):
-                shutil.move(f, os.path.join(output_include_dir, os.path.basename(f)))
-            for f in glob.glob(os.path.join(container_output_dir, "*.c")):
-                shutil.move(f, os.path.join(output_source_dir, os.path.basename(f)))
-
-    # adjustIncludes(output_include_dir)
-    modifyIncludes(output_include_dir)
-    modifyIncludes(output_source_dir)
-
-    print(f"Applying patches ...")
-    script_dir = os.path.dirname(os.path.abspath(__file__))
-    patch_file = os.path.join(script_dir, f"patches/{args.type}.patch")
-    if os.path.exists(patch_file):
-        subprocess.run(["git", "apply", patch_file], check=True)
-
-    print(f"Generated C/C++ library for {args.type}")
+            spec_file= os.path.basename(asn1_file)
+            spec_name=spec_file.split(".")[0]
+
+    
+            # create output directories
+            output_dir = os.path.realpath(args.output_dir)
+            output_include_dir = os.path.join(args.output_dir,spec_name,"include")
+            output_source_dir = os.path.join(args.output_dir,spec_name,"src")
+            os.makedirs(output_include_dir, exist_ok=True)
+            os.makedirs(output_source_dir, exist_ok=True)
+
+            # create temporary directories for running asn1c in docker container
+            with tempfile.TemporaryDirectory() as temp_input_dir:
+                with tempfile.TemporaryDirectory() as temp_output_dir:
+
+                    if args.temp_dir is None:
+                        container_input_dir = temp_input_dir
+                        container_output_dir = temp_output_dir
+                        asn1c_cmd_file = tempfile.NamedTemporaryFile(delete=False).name
+                    else:
+                        container_input_dir = os.path.join(args.temp_dir, "input")
+                        container_output_dir = os.path.join(args.temp_dir, "output")
+                        os.makedirs(container_input_dir, exist_ok=True)
+                        os.makedirs(container_output_dir, exist_ok=True)
+                        asn1c_cmd_file = os.path.join(os.path.dirname(os.path.abspath(__file__)), "asn1c.sh")
+
+
+                    # copy input asn1 files to temporary directory
+                    
+                    shutil.copy(asn1_file, container_input_dir)
+
+                    # run asn1c docker container to generate header and source files
+                    with open(asn1c_cmd_file, "w") as f:
+                        f.write(f"asn1c $(find /input -name '*.asn1' | sort) -fcompound-names  -no-gen-BER -no-gen-XER -no-gen-JER -no-gen-OER -no-gen-example -gen-UPER")
+                    
+                    subprocess.run(["docker", "run", "--rm", "-u", f"{os.getuid()}:{os.getgid()}", "-v", f"{container_input_dir}:/input:ro", "-v", f"{container_output_dir}:/output", "-v", f"{asn1c_cmd_file}:/asn1c.sh", args.docker_image], check=True)
```

Comment:
This seems a bit weird to me. Is this script not already running inside a container? As I understood it, the code in this repo should build all into it's own docker image -- with all associated dependencies, including the etsi-its-message packages -- that the user then runs via `docker run...` execute this code and produce the output. This seems like this script is run outside of docker, on the host, and **it** calls `docker run...`?

I'm not sure that's a sensible approach, as it means that the dependencies for this script (and the rest of the code that might run outside the container, still need to be installed on the host. Or am I misunderstanding what's happening here?